### PR TITLE
simple (ticdc): support send all tables bootstrap message at changefeed start 

### LINKED ISSUE #11315: Simple protocol: Send all Bootstrap when changefeed starts
### Is your feature request related to a problem?

Some user requires the schema of all tables, which the changefeed synchronizes, to be sent downstream as Bootstrap events when the changefeed begins. 

To fulfill this requirement, a boolean parameter send-all-bootstrap-at-start can be added, which is set to false by default.
If this parameter is set to true, the changefeed, upon starting, will send the schema of all synchronized tables downstream as Bootstrap events. This will be done synchronously in the ddlSink. After the Bootstrap events are synchronized, row changes will begin to be synchronized. 

### Describe the feature you'd like

as above.

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #11315 

### What is changed and how it works?
A boolean parameter `send-all-bootstrap-at-start` is added, which is set to false by default.
If this parameter is set to true, the changefeed, upon starting, will send the schema of all synchronized tables downstream as Bootstrap events. This will be done synchronously in the ddlSink. After the Bootstrap events are synchronized, row changes will begin to be synchronized. 

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/owner/ddl_sink.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ type DDLSink interface {
 	// the caller of this function can call again and again until a true returned
 	emitDDLEvent(ctx context.Context, ddl *model.DDLEvent) (bool, error)
 	emitSyncPoint(ctx context.Context, checkpointTs uint64) error
+	emitBootstarp(ctx context.Context, bootstrap *model.DDLEvent) error
```
Comment (hongyunyan) [inline cdc/owner/ddl_sink.go:57]: It seems a typo issue: bootstarp --> bootstrap?
BTW, Please add a comment for this function

File: cdc/owner/ddl_manager.go [hunk 2]
```diff
@@ -130,6 +131,9 @@ type ddlManager struct {
 
 	BDRMode       bool
 	ddlResolvedTs model.Ts
+
+	sinkConfig  *config.SinkConfig
```
Comment (hongyunyan) [inline cdc/owner/ddl_manager.go:135]: We just use the m.sinkConfig.ShouldSendAllBootstrapAtStart() in ddlManager, could we just store this parameter, not all the sinkConfig.

It seems a little wired that ddlManager has a member call sinkConfig.

File: cdc/owner/ddl_sink.go (hunk 1 above)
Comment (3AceShowHand) [reply cdc/owner/ddl_sink.go:57]: It looks that `emitBootstrap` is almost the same as `emitDDLEvent`.

IMPO, they should share the same interface method.

**Review by 3AceShowHand — APPROVED** [review]

File: cdc/owner/ddl_sink.go (hunk 1 above)
Comment (asddongmen) [reply cdc/owner/ddl_sink.go:57]: emitBootstrap is a blocking interface, whereas emitDDLEvent is an asynchronous interface.

File: cdc/owner/ddl_sink.go [hunk 3]
```diff
@@ -54,6 +54,7 @@ type DDLSink interface {
 	// the caller of this function can call again and again until a true returned
 	emitDDLEvent(ctx context.Context, ddl *model.DDLEvent) (bool, error)
 	emitSyncPoint(ctx context.Context, checkpointTs uint64) error
+	emitBootstrap(ctx context.Context, bootstrap *model.DDLEvent) error
```
Comment (hongyunyan) [inline cdc/owner/ddl_sink.go:57]: Could you add some comments for emitBootStrap? So that we can quickly know the function usage.

**Review by hongyunyan — APPROVED** [review]
rest is LGTM

**Review by benmeadowcroft — APPROVED** [review]

**Pull request merged**